src/components/calendar/CalendarList.tsx

```diff
@@ -0,0 +1,155 @@
+import { useTranslation } from 'language/LanguageProvider';
+import React, { useEffect, useState } from 'react';
+import { FlatList} from 'react-native';
+import { Divider, Title } from 'react-native-paper';
+
+import { Calendar, entryGt } from 'storage/calendar';
+import { CalendarListSection } from './CalendarListSection';
+
+/**
+ * Returns a new calendar grouped by date.
+ *
+ * @remarks
+ * This function assumes the calendar is sorted.
+ *
+ * @param calendar - The calendar to group by date (day)
+ * @returns A list of calendars where each calendar is a day of entries
+ */
+const groupByDate = (calendar: Calendar): Calendar[] => {
+  let groups: Calendar[] = [];
+  let currentGroup: Calendar = [];
+  let date = "";
+
+  calendar.forEach((entry) => {
+    if (entry.date == date) {
+      currentGroup.push(entry);
+    } else {
+      if (currentGroup.length !== 0){
+        groups.push(currentGroup);
+      }
+      currentGroup = [entry];
+      date = entry.date;
+    }
+  });
+
+  if (currentGroup.length !== 0) {
+    groups.push(currentGroup);
+  }
+
+  return groups;
+}
+
+/**
+ * Inserts headers of the form YYYY-mm before calendars that has the
+ * header as year and month.
+ *
+ * @remarks
+ * This function assumes the calendar is sorted.
+ *
+ * @param groups - The list of calendars to insert headers into
+ * @returns A list containing calendars and strings (headers)
+ */
+const insertMonthHeaders = (groups: Calendar[]): (Calendar | string)[] => {
+  let groupsWithHeaders: (Calendar | string)[] = [];
+  let currentMonth = "";
+
+  groups.forEach(group => {
+    const month = group[0].date.slice(0, 7); // YYYY-mm
+    if (month !== currentMonth) {
+      groupsWithHeaders.push(month);
+    }
+    groupsWithHeaders.push(group);
+    currentMonth = month;
+  });
+
+  return groupsWithHeaders;
+}
+
+/**
+ * Returns the current date in ISO format. Normal programming languages use
+ * something like date.strftime("%Y-%m-%d").
+ *
+ * @returns A ISO-formatted date
+ */
+const ISODate = (): string => {
+  const date = new Date();
+  const year = date.getFullYear();
+  let month = '' + (date.getMonth() + 1);
+  let day = '' + date.getDate();
+
+  if (month.length < 2) {
+    month = '0' + month;
+  }
+  if (day.length < 2) {
+    day = '0' + day;
+  }
+
```

Comment:
So I looked at it and found no easy way to get the format we want (you could argue this is the easiest way). Going down the rabbit hole of dates and locales is not very high on the list right now.

Reply:
It was just a suggestion if you thought it was easier. I think the way you do it is good as well

Reply:
So the reason why i think Malin brought this up is because we are already using this module due to it being required by one of the new dependencies in #18. Using this module we would allow for formatted localizations of dates, numbers, times, etc, and i would argue that this feature is vital for the app since its a major part of localization.

Regarding this specific case a string containing the year, month and day should be retrieved with something similar to this:
`Intl.DateTimeFormat(settings.language).format(date)` (Does not require any imports)

There are many other properties that could be specified but that should not be required in this case. 

I'll redirect any further discussion about this #18 since the android specific library support is implemented there. The code here can stay as it is but if we proceed as planned in #18 i think this component should reflect and utilize the Intl functionality.

Reply:
There is a good reason why Intl is not used here

https://stackoverflow.com/questions/29961992/javascript-intl-datetimeformatter-as-iso-8601custom-output-pattern
> ISO-8601 format is not something that's guaranteed to be the appropriate format for any particular locale, so you can't use Intl to produce it.

I think it is possible to create the right format, but it seems like a roundabout way of doing it. If I understand it correctly, Intl is used for creating time strings which are displayed to users, which is not what it is doing here.

In this file it is used to compare starting times of calendar entries since dates are stored as ISO dates (strings).